jwcrypto/jwk.py

```diff
@@ -243,13 +254,10 @@ def _encode_int(self, i):
 
     def _generate_RSA(self, params):
         pubexp = 65537
-        size = 2048
+        size = self._get_gen_size(params, 2048)
         if 'public_exponent' in params:
             pubexp = params['public_exponent']
             del params['public_exponent']
```

Comment:
pubexp = params.pop('public_exponent') gets and deletes the entry from the dict at the same time